crates/ursa-gateway/src/server/route/api/v1/get.rs

```diff
@@ -84,6 +87,92 @@ pub async fn get_block_handler(
 
     debug!("received indexer response for {cid}:\n{indexer_response:?}");
 
+    let provider_resp = match choose_provider(indexer_response) {
+        Ok(addrs) => {
+            let mut addr_iter = addrs.into_iter();
+
+            loop {
+                match addr_iter.next() {
+                    Some(addr) => {
+                        let req = Request::builder()
+                            .method("POST")
+                            .uri(format!("http://{}:{}/rpc/v0", addr.ip(), addr.port()))
+                            .header("Content-Type", "application/json")
+                            .body(Body::from(
+                                json!({
+                                    "jsonrpc": "2.0",
+                                    "id": "id",
+                                    "method": "ursa_get_cid",
```

Comment:
this method only gets a single block. is this what this handler is trying to do, to just get a single block?
if it is meant to cache/get the full content, it should call the stream(http) api

Reply:
@theBeardA I believe this is only to get a block per the handler function name and the todo comments but I could be wrong. @scott-dn is it?

Reply:
@kckeiks as the gateway, we want full content, but that needed a streaming handler which we need atm (#198).
For this PR, let's assume we only get a single block.
hope it makes sense to you @theBeardA .

Reply:
sure, we wanted to remove that method, since it's kinda useless, once we have more rpc methods like getting blockchain info. just wanted to make note of that

Reply:
Good to know. Then it sounds like we def need a way to pipe the responses then.